--- postfixadmin/languages/pl.lang.fix	2007-04-30 07:43:28.270208911 +0200
+++ postfixadmin/languages/pl.lang	2007-04-30 08:20:50.113964306 +0200
@@ -39,12 +39,12 @@
 $PALANG['pMain_viewlog'] = 'Poka¿ pliki logów.';
 $PALANG['pMain_logout'] = 'Zakoñcz pracê z systemem';
 
-$PALANG['pOverview_disabled'] = 'Disabled';
-$PALANG['pOverview_unlimited'] = 'Unlimited';
-$PALANG['pOverview_title'] = ':: Defined Domains';
-$PALANG['pOverview_up_arrow'] = 'Go Top';
-$PALANG['pOverview_right_arrow'] = 'Next Page';
-$PALANG['pOverview_left_arrow'] = 'Previus Page';
+$PALANG['pOverview_disabled'] = 'Wy³±czony';
+$PALANG['pOverview_unlimited'] = 'Nieograniczony';
+$PALANG['pOverview_title'] = ':: Zdefiniowane domeny';
+$PALANG['pOverview_up_arrow'] = 'Pocz±tek listy';
+$PALANG['pOverview_right_arrow'] = 'Nastêpna Strona';
+$PALANG['pOverview_left_arrow'] = 'Poprzednia Strona';
 $PALANG['pOverview_alias_title'] = ':: Alias';
 $PALANG['pOverview_mailbox_title'] = ':: Mailboxes';
 $PALANG['pOverview_button'] = 'Id¼';
@@ -80,7 +80,7 @@
 $PALANG['pCreate_alias_goto_text_error'] = 'Gdzie ten email powinien dotrzeæ.<br /><span class="error_msg">The TO is not valid!</span>';
 $PALANG['pCreate_alias_result_error'] = '<span class="error_msg">Nie mo¿na dodaæ aliasu do tabeli aliasów!</span>';
 $PALANG['pCreate_alias_result_succes'] = 'Alias zosta³ dodany do tabeli aliasów!';
-$PALANG['pCreate_alias_catchall_text'] = 'Aby utworzyæ domy¶lne konto dla domeny (catch-all) podaj "*" (gwiazdkê) jako alias.<br />Je¶li chcesz przekazywaæ ca³o¶æ poczty do innej domeny, wpisz jako alias "*@domena.tld".';
+$PALANG['pCreate_alias_catchall_text'] = 'Aby utworzyæ domy¶lne konto dla domeny (catch-all) podaj "*" (gwiazdkê) jako alias.<br />Je¶li chcesz przekazywaæ ca³o¶æ poczty do innej domeny, wpisz "*@domena.tld" jako Do.';
 $PALANG['pEdit_alias_welcome'] = 'Edytuj alias dla Twojej domeny.<br />Jeden wpis na liniê.';
 $PALANG['pEdit_alias_address'] = 'Alias';
 $PALANG['pEdit_alias_address_error'] = '<span class="error_msg">Nie mo¿na zokalizowaæ aliasu!</span>';
@@ -158,13 +158,13 @@
 $PALANG['pSendmail_result_error'] = '<span class="error_msg">Nie mo¿na utworzyæ konta!</span>';
 $PALANG['pSendmail_result_succes'] = 'Konto zosta³o utworzone!';
 
-$PALANG['pAdminMenu_list_admin'] = 'Lista administratorów';
-$PALANG['pAdminMenu_list_domain'] = 'Lista domen';
+$PALANG['pAdminMenu_list_admin'] = 'Administratorzy';
+$PALANG['pAdminMenu_list_domain'] = 'Domeny';
 $PALANG['pAdminMenu_list_virtual'] = 'Virtual List';
 $PALANG['pAdminMenu_viewlog'] = 'Logi';
-$PALANG['pAdminMenu_backup'] = 'Kopia bezpieczeñstwa';
+$PALANG['pAdminMenu_backup'] = 'Backup';
 $PALANG['pAdminMenu_create_domain_admins'] = 'Administratorzy domeny';
-$PALANG['pAdminMenu_create_admin'] = 'Nowy administrator';
+$PALANG['pAdminMenu_create_admin'] = 'Nowy admin';
 $PALANG['pAdminMenu_create_domain'] = 'Nowa domena';
 $PALANG['pAdminMenu_create_alias'] = 'Dodaj alias';
 $PALANG['pAdminMenu_create_mailbox'] = 'Dodaj konto';
@@ -186,7 +186,7 @@
 $PALANG['pAdminList_domain_active'] = 'Aktywne';
 
 $PALANG['pAdminList_virtual_button'] = 'Id¼';
-$PALANG['pAdminList_virtual_welcome'] = 'Overview for ';
+$PALANG['pAdminList_virtual_welcome'] = 'Przegl±d ';
 $PALANG['pAdminList_virtual_alias_alias_count'] = 'Aliasy';
 $PALANG['pAdminList_virtual_alias_mailbox_count'] = 'Konta';
 $PALANG['pAdminList_virtual_alias_address'] = 'Od';
@@ -276,22 +276,22 @@
 $PALANG['pUsersVacation_subject_text'] = 'Poza biurem';
 $PALANG['pUsersVacation_body'] = 'Tekst';
 $PALANG['pUsersVacation_body_text'] = <<<EOM
-I will be away from <date> until <date>.
-For urgent matters you can contact <contact person>.
+Nie bêdzie mnie od <daty> do <daty>.
+W pilnej sprawie proszê kontaktowaæ siê z <osob± kontaktow±>.
 EOM;
 $PALANG['pUsersVacation_button_away'] = 'Nieobecny/a';
-$PALANG['pUsersVacation_button_back'] = 'Zaraz wracam';
+$PALANG['pUsersVacation_button_back'] = 'Wróci³em';
 $PALANG['pUsersVacation_result_error'] = '<span class="error_msg">Nie mogê zaktualizowaæ ustawieñ Twojej auto odpowiedzi!</span>';
 $PALANG['pUsersVacation_result_succes'] = 'Twoja auto odpowied¼ zosta³a usuniêta!';
 
-$PALANG['pCreate_dbLog_createmailbox'] = 'create mailbox';     
-$PALANG['pCreate_dbLog_createalias'] = 'create alias';
-$PALANG['pDelete_dbLog_deletealias'] = 'delete alias';
-$PALANG['pDelete_dbLog_deletemailbox'] = 'delete mailbox';
+$PALANG['pCreate_dbLog_createmailbox'] = 'utwórz skrzynkê pocztow±';
+$PALANG['pCreate_dbLog_createalias'] = 'utwórz alias';
+$PALANG['pDelete_dbLog_deletealias'] = 'usuñ alias';
+$PALANG['pDelete_dbLog_deletemailbox'] = 'usuñ skrzynkê pocztow±';
 
 $PALANG['pEdit_dbLog_editactive'] = 'change active state';
-$PALANG['pEdit_dbLog_editalias'] = 'edit alias';
-$PALANG['pEdit_dbLog_editmailbox'] = 'edit mailbox';
+$PALANG['pEdit_dbLog_editalias'] = 'edytuj alias';
+$PALANG['pEdit_dbLog_editmailbox'] = 'edytuj skrzynkê pocztow±';
 
 $PALANG['pSearch_welcome'] = 'Searching for: ';
 ?>
