--- ./users/edit-alias.php.org	2008-07-10 10:00:44.059360124 +0200
+++ ./users/edit-alias.php	2008-07-10 10:01:27.996021987 +0200
@@ -61,10 +61,10 @@ if ($_SERVER['REQUEST_METHOD'] == "POST"
    $fGoto = strtolower ($fGoto);
 
    $goto = preg_replace ('/\\\r\\\n/', ',', $fGoto);
-	$goto = preg_replace ('/\r\n/', ',', $fGoto);
-	$goto = preg_replace ('/[\s]+/i', '', $goto);
-	$goto = preg_replace ('/\,*$/', '', $goto);
-	$array = preg_split ('/,/', $goto);
+   $goto = preg_replace ('/\r\n/', ',', $goto);
+   $goto = preg_replace ('/[\s]+/i', '', $goto);
+   $goto = preg_replace ('/\,*$/', '', $goto);
+   $array = preg_split ('/,/', $goto);
 
 	for ($i = 0; $i < sizeof ($array); $i++) {
 		if (in_array ("$array[$i]", $CONF['default_aliases'])) continue;
